centml/compiler/server.py

```diff
@@ -20,7 +20,10 @@ def get_status(model_id: str):
     if not os.path.isdir(os.path.join(config_instance.SERVER_BASE_PATH, model_id)):
         return CompilationStatus.NOT_FOUND
 
-    if not os.path.isfile(get_server_compiled_forward_path(model_id)):
+    # For the case where size == 0, torch.save creates a zip file before saving the actual data.
+    # We don't want this empty zipfile to be considered the serialized forward function
+    save_path = get_server_compiled_forward_path(model_id)
+    if not os.path.isfile(save_path) or os.path.getsize(save_path) == 0:
```

Comment:
what if the file is half way through the writing? Then the size won't be 0 but the file is still incomplete. 

Is the behaviour of torch.save(), it first write an empty file and then dumps the whole data into to the file in a single shot?

Reply:
That's a good point. It creates the zipfile, then writes to it multiple times, not just in a single shot.

I'll implement some mutexes to avoid this race condition unless you have a better approach in mind

Reply:
writing our own synchronization logic might be painful. I would suggest to first check if there is a way to write the entire file atomically.

if not, we can use our own locking mechanism.

Reply:
what do you think about writing the zipfile to the cache under a different cache (that /status/ doesn';t check for) and then changing the name once we finish writing to it?